src/realm-settings/user-profile/attribute/AddValidatorDialog.tsx

```diff
@@ -24,13 +24,19 @@ export const AddValidatorDialog = ({
 }: AddValidatorDialogProps) => {
   const { t } = useTranslation("realm-settings");
   const [selectedValidator, setSelectedValidator] = useState<Validator>();
+  const [validators, setValidators] = useState(allValidator);
   const [addValidatorRoleModalOpen, toggleModal] = useToggle();
 
   return (
     <>
       {addValidatorRoleModalOpen && (
         <AddValidatorRoleDialog
-          onConfirm={(newValidator) => onConfirm(newValidator)}
+          onConfirm={(newValidator) => {
+            onConfirm(newValidator);
+            setValidators(
+              validators.filter((v) => v.name !== newValidator.name)
```

Comment:
You could use some destructuring here.
```suggestion
              validators.filter(({ name }) => name !== newValidator.name)
```